### ISSUE TITLE
Need clean way to share data between parameterized tests

### ISSUE THREAD

**Issue by JVimes** [Microsoft/testfx#169]
MSTest v2 needs a clean way to share test-data between tests. It should support objects.

Current approaches fall short:
* DataRow can't be shared and doesn't support objects (just compile-time values).
* DataSource doesn't support objects, and requires a database/csv.
* Looping through an array/collection lumps the cases into a single test, making test results less readable.

Suggest adding the equivalent of NUnit's [TestCaseSource](https://www.nunit.org/index.php?p=testCaseSource&r=2.5.9), which takes the name of a field, property, or method that returns an IEnumerable (which can handle objects):

```c#
static object[] DivideCases =
{
    new object[] { 12, 3, 4 },
    new object[] { 12, 2, 6 },
    new object[] { 12, 4, 3 } 
};

[Test, TestCaseSource("DivideCases")]
public void DivideTest(int n, int d, int q)
{
    Assert.AreEqual( q, n / d );
}
```

**Comment by AbhitejJohn** [comment]
Absolutely. Thanks for filing this @JVimes . This should also help drive a structure around #141.

**Comment by JVimes** [comment]
This is being discussed in: Microsoft/testfx-docs#12

**Comment by AbhitejJohn** [comment]
This has been addressed in 1.2.0-beta release of the Framework/Adapter pair.

**Comment by JVimes** [comment]
Thanks! For others who find this, here's how to try it out:

Make sure you have MSTest v1.2.0 or later. At time of writing it's in beta, so need to install yourself using NuGet's "Include prerelease" checkbox:

![image](https://user-images.githubusercontent.com/6148201/28543395-d2b3db40-707c-11e7-88c9-2b0e99cd6805.png)

Then, add a static, `object[][]` property to your test class. Fill in data. Add`[DynamicData("ThePropertyName")]` to your method, and make sure it has the right number/type of parameters:

```C#
static object[] Data
{
    get => new[]
    {
        new object[] {1, 2, 3},
        new object[] {4, 5, 6}
    };
}
        
[TestMethod]
[DynamicData("Data")]
public void Test1(int a, int b, int c)
{
    Assert.AreEqual(1, a % 3);
    Assert.AreEqual(2, b % 3);
    Assert.AreEqual(0, c % 3);
}
```

You can alternatively use a method that returns an IEnumerable<object[]>, instead of a property. I'm going off [this doc](https://github.com/Microsoft/testfx-docs/pull/12/files?short_path=be0f4e0#diff-be0f4e0458235b8e2b4711f96924ea84).
